crates/cli/src/config/mod.rs

```diff
@@ -0,0 +1,97 @@
+use serde::{Deserialize, Serialize};
+use std::fs;
+
+// ! this is the struct that will be used to read the config.toml table
+#[derive(Serialize, Deserialize, Debug)]
+struct ConfigToml {
+    network: Option<ConfigTomlNetwork>,
+    see: Option<ConfigTomlSee>,
+}
+// ! this is the struct that will be used to read the network config.toml values
+#[derive(Serialize, Deserialize, Debug)]
+struct ConfigTomlNetwork {
+    rpc_url: Option<String>,
+}
+// ! this is the struct that will be used to read the see config.toml values
+#[derive(Serialize, Deserialize, Debug)]
+struct ConfigTomlSee {
+    token0: Option<String>,
+    token1: Option<String>,
+    bp: Option<String>,
+}
+// ! this is the struct that will be used to represent the config.toml values and is returned
+#[derive(Debug)]
+pub struct Config {
+    pub rpc_url: String,
+    pub token0: String,
+    pub token1: String,
+    pub bp: String,
+}
+
+impl Config {
+    // ! when makin a new instance of the config object we need to read the config.toml file and then return the values
+    pub fn new() -> Self {
+        let config_filepaths: [&str; 2] = [
+            "./crates/cli/src/config.toml",
+            "./crates/cli/src/Config.toml",
+        ];
+
+        let mut content: String = "".to_owned();
+        // ! Read from list of filepaths until we find a file that exists
+        for filepath in config_filepaths {
+            let result = fs::read_to_string(filepath);
+            match result {
+                Ok(c) => content = c,
+                Err(_) => break,
+            }
+        }
+
+        // ! initialize the outer struct with values from the config.toml file
+        let config_toml: ConfigToml = toml::from_str(&content).unwrap_or_else(|_| {
+            println!("Failed to create ConfigToml object from config file");
+            ConfigToml {
+                network: None,
+                see: None,
+            }
+        });
+        // ! check the rpc_url field in the network config.toml section
+        let rpc_url = match config_toml.network {
+            Some(network) => network.rpc_url.unwrap_or_else(|| {
+                println!("Missing Field rpc_url in network config");
+                "unknown".to_owned()
+            }),
+            None => "unknown".to_owned(),
```

Comment:
Setting the field to `"unknown"` is basically just letting the program panic at a later stage so its basically redundant here imo

Reply:
This would be pretty good imo